cpp/include/cudf/detail/utilities/trie.cuh

```diff
@@ -135,24 +135,22 @@ inline thrust::host_vector<SerialTrieNode> createSerializedTrie(
  * @return Boolean value, true if string is found, false otherwise
  */
 __host__ __device__ inline bool serialized_trie_contains(device_span<SerialTrieNode const> trie,
-                                                         char const *key,
-                                                         size_t key_len)
+                                                         device_span<char const> key)
 {
   if (trie.data() == nullptr || trie.empty()) return false;
-  if (key_len == 0) return trie[0].is_leaf;
-  int curr_node = 1;
-  for (size_t i = 0; i < key_len; ++i) {
+  if (key.empty()) return trie.front().is_leaf;
```

Comment:
Outside the scope of this PR but are we able to change `device_span::empty` to `device_span::is_empty`?

Reply:
The span classes are meant to be equivalent to `std::span` (or `thrust::span`). I don't think we want to deviate from the API, it would make eventual swap to standard version much more difficult.
CC @cwharris

Reply:
Sad the committee got this wrong.